src/wallet/rpcwallet.cpp

```diff
@@ -3324,6 +3475,125 @@ UniValue z_getbalance(const UniValue& params, bool fHelp)
     return ValueFromAmount(nBalance);
 }
 
+UniValue z_getbalanceforaddress(const UniValue& params, bool fHelp)
+{
+    if (!EnsureWalletIsAvailable(fHelp))
+        return NullUniValue;
+
+    if (fHelp || params.size() < 1 || params.size() > 2)
+        throw runtime_error(
+            "z_getbalanceforaddress \"address\" ( minconf )\n"
+            "\nReturns the per-pool balances of a Unified Address belonging to the node's wallet."
+            "\nArguments:\n"
+            "1. \"address\"      (string) The selected address. It may be a transparent, shielded, or Unified address.\n"
+            "2. minconf        (numeric, optional, default=1) Only include transactions confirmed at least this many times.\n"
+            "\nResult:\n"
+            "{\n"
+            "  \"pool_balances\": {\n"
```

Comment:
I personally think that the `_balances` suffix is redundant, and we should use the same key as we do in other places we refer to a dictionary of data about the shielded and transparent pools:
```suggestion
            "  \"pools\": {\n"
```
I'm willing to be convinced that we should use `pool_balances` here if someone else feels strongly about it.

Reply:
`pools` seems fine to me.